Doc/library/unittest.mock.rst

```diff
@@ -1099,6 +1101,58 @@ object::
       [call('foo'), call('bar')]
 
 
+.. class:: ThreadingMock(spec=None, side_effect=None, return_value=DEFAULT, wraps=None, name=None, spec_set=None, unsafe=False, **kwargs)
+
+  A version of :class:`MagicMock` for multithreading tests. The
+  :class:`ThreadingMock` object provides extra methods to wait for a call to
+  happen on a different thread, rather than assert on it immediately.
+
+  .. method:: block_until_called()
+
+      Blocks until the the mock is called.
+
+        >>> mock = ThreadingMock()
+        >>> thread = threading.Thread(target=mock)
+        >>> thread.start()
+        >>> mock.wait_until_called(1)
```

Comment:
I find this pretty hard to read... what is `1` here? I have to go dig...
```
 >>> mock.wait_until_called(timeout=1)
```
...is clearer, but then we have the name clash.

How about:

```
>>> mock.until_called().wait(timeout=1)
```

Reply:
`mock.until_called().wait(timeout=1)`

What does `mock.until_called()` do? It worries me that the syntax is quite different from the rest of the mock methods.

Reply:
> I find this pretty hard to read... what is 1 here? I have to go dig...

Absolutely agree, which is why I was pushing back on it initially. I'd expect most users to either use `block_until_called` or `wait_until_called` with a single timeout for all method. Two ideas I can think of are:

1. Document `wait_until_called` with a constant:
```python
>>> TIMEOUT = 1
>>> mock.wait_until_called(TIMEOUT)
```
2. Find a way to inject this in a different place (I was thinking on the `ThreadingMock` constructor (but again, risk of clash).
```python
>>> mock = ThreadingMock(timeout=1)
>>> mock.wait_until_called()
```

Reply:
I like option 2 here, can you give me an example of a clash when using `ThreadingMock(timeout=1)`?

Reply:
> What does `mock.until_called()` do? It worries me that the syntax is quite different from the rest of the mock methods.

Yeah, I definitely don't love it ;-) `mock.until_called()` would return an object that you can call `wait()` on, nothing more. It's just syntactic sugar to avoid the potential argument name clash. it sort-of has equivalents in the way you set side effect and return values through the existing API, but only if you squint in a certain way...

Reply:
> I like option 2 here, can you give me an example of a clash when using ThreadingMock(timeout=1)?

Because I was thinking that we'd need to expose that in a property and it could clash with the mocked object properties, but it actually doesnt! We could just take it as the constructor and not expose it in the class. Let me work on a proposal.